plugins/depthCamera/src/DepthCameraDriver.cpp

```diff
@@ -161,19 +170,28 @@ bool GazeboYarpDepthCameraDriver::OnNewImageFrame(const unsigned char *_image,
                           unsigned int _width, unsigned int _height,
                           unsigned int _depth, const std::string &_format)
 {
-
 //     std::cout << "OnNewImageFrame: " <<
 //                     "\n\t_width is " << _width    <<
 //                     "\n\t_height is " << _height   <<
 //                     "\n\t_depth is " << _depth     <<
 //                     "\n\t_format is " << _format   << std::endl;
 
+
+#if GAZEBO_MAJOR_VERSION >= 7
+    m_colorFrameMutex.wait();
+    if(m_depthCameraSensorPtr->IsActive())
+        memcpy(m_imageFrame_Buffer, m_depthCameraPtr->ImageData(), m_imageFrame_BufferSize);
+
+    m_colorTimestamp.update(this->m_depthCameraSensorPtr->LastUpdateTime().Double());
+    m_colorFrameMutex.post();
+#else
     m_colorFrameMutex.wait();
```

Comment:
This mutex wait/post can go outside the #if blocks (it is shared between the two branches).